packages/flutter_tools/lib/src/cache.dart

```diff
@@ -1487,3 +1387,148 @@ const List<List<String>> _dartSdks = <List<String>> [
   <String>['linux-x64', 'dart-sdk-linux-x64.zip'],
   <String>['windows-x64', 'dart-sdk-windows-x64.zip'],
 ];
+
+/// An API for downloading and un-archiving artifacts, such as engine binaries or
```

Comment:
Up to you, but I'm wondering if it makes sense to pull this out under `base/`

Reply:
I think at this point it is too tied to the cache API. I think it might make sense to consider combining this with the `Net` class in the future, at which point it will be more coherent and `base/` will make more sense IMO